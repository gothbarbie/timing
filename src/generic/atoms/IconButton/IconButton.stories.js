import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { library } from '@fortawesome/fontawesome-svg-core'
import {
  faArrowCircleLeft,
  faArrowCircleRight,
  faCog,
  faEllipsisH,
  faTimesCircle,
  faPlusCircle,
} from '@fortawesome/free-solid-svg-icons'

import IconButton from './IconButton'

library.add(
  faArrowCircleLeft,
  faArrowCircleRight,
  faCog,
  faEllipsisH,
  faPlusCircle,
  faTimesCircle
)

storiesOf('IconButton', module)
  .add('default', () => <IconButton icon={['fas', 'cog']} />)
  .add('clickable', () => (
    <IconButton icon={['fas', 'cog']} onClick={action('clicked')} />
  ))
