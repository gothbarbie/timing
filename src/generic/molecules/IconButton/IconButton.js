import React from 'react'
import styled from 'styled-components'

import Icon from '../../atoms/Icon'

const Button = styled.button`
  padding: 0;
  margin: 0;
  outline: none;
  background-color: transparent;
  border: none;
`

const IconButton = ({ color, icon, size = 'lg', onClick }) => (
  <button onClick={onClick}>
    <Icon color={color} icon={icon} size={size} />
  </button>
)

export default IconButton
