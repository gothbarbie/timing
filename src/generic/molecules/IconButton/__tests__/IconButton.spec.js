import React from 'react'
import { shallow } from 'enzyme'

import IconButton from '../IconButton'

describe('generic/molecules', () => {
  it('renders', () => {
    const component = shallow(<IconButton />)

    expect(component).toMatchSnapshot()
  })
})
