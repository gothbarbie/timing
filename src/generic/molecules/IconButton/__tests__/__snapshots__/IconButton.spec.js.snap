// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic/molecules renders 1`] = `
<button>
  <Icon
    size="lg"
  />
</button>
`;
